### PR TITLE
Change `launch` to `Launcher` dataclass

### PR DESCRIPTION
**Pull request by apoorvkh**

### CODE REVIEW

**Comment by pmcurtin** [comment]
CI test will fail because it uses `torchrunx.launch`

**Comment by apoorvkh** [comment]
todo: add alias for `launch()` with pytest to ensure arguments are the same as `Launcher().run()` and do something clever to copy the argument docstrings

**Comment by apoorvkh** [comment]
we can probably still feature `trx.launch()` in the README as the easy way to use this ... 
but in the "advanced usage", we can mention the `Launcher` class for CLI purposes

**Review by pmcurtin — APPROVED** [review]
lgtm!

File: src/torchrunx/slurm.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import subprocess
 
 
-def slurm_hosts() -> list[str]:
+def slurm_hosts() -> "list[str]":
```
Comment (apoorvkh) [inline src/torchrunx/slurm.py:5]: Wait, why this change?

Comment (pmcurtin) [reply src/torchrunx/slurm.py:5]: Pyright was complaining about it.
